### PR TITLE
Added create folder button to project imports

### LINKED ISSUE #2754: Add create folder button to project imports
### Describe the project you are working on
Various projects downloaded from Git.

### Describe the problem or limitation you are having in your project
Importing projects from a zip or project file in the import menu of the project manager requires users to create an empty folder outside of the godot editor. This adds extra time that can build up while checking out many different projects.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a "Create Folder" button to the import existing project menu that will parse the name of the file to import and create a folder for it in the same way as creating a new project gives you the option to create a folder.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

These are screenshots from a branch I have where this feature is implemented.

Screenshot from 2021-05-21 17-23-24
Screenshot from 2021-05-21 17-23-52

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It requires engine changes.

### Is there a reason why this should be core and not an add-on in the asset library?

It's an editor-specific function.

### PR DESCRIPTION
**Pull request by likeich**
Closes https://github.com/godotengine/godot-proposals/issues/2754

This adds a "create folder" button to the import project window in the project manager as described in the linked proposal.

### CODE REVIEW

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -376,20 +376,33 @@ class ProjectDialog : public ConfirmationDialog {
...
-	void _create_folder() {
-		const String project_name_no_edges = project_name->get_text().strip_edges();
+	void _create_new_project_folder() {
+		_create_folder(project_name->get_text(), project_path);
+	}
+
+	void _create_install_folder() {
+		Vector<String> strings = project_path->get_text().split("/");
+		String filename = strings[strings.size() - 1];
+		String filename_no_extension = filename.split(".")[0];
```
Comment (Calinou) [inline editor/project_manager.cpp:386]: If the file name has multiple periods (e.g. `project.sample.zip`), this means the file name would be considered to be `project`, which is incorrect.

Instead, you can use `filename.rstrip(filename.get_extension())`.

Comment (likeich) [reply editor/project_manager.cpp:386]: Fixed, I'll test and then amend the commit.

File: editor/project_manager.cpp [hunk 2]
```diff
@@ -376,20 +376,33 @@ class ProjectDialog : public ConfirmationDialog {
...
-	void _create_folder() {
-		const String project_name_no_edges = project_name->get_text().strip_edges();
+	void _create_new_project_folder() {
+		_create_folder(project_name->get_text(), project_path);
+	}
+
+	void _create_install_folder() {
+		Vector<String> strings = project_path->get_text().split("/");
+		String filename = strings[strings.size() - 1];
+		String filename_no_extension = filename.trim_suffix("." + filename.get_extension());
```
Comment (likeich) [inline editor/project_manager.cpp:386]: I added the `"." +` because `get_extension()` doesn't return the "." of the extension.

Comment (nathanfranke) [reply editor/project_manager.cpp:386]: `filename.get_basename()` is better in this case, although note that I have a PR that supersedes this

**Comment by akien-mga** [comment]
Superseded by #56420. Thanks for the contribution!

**Pull request closed without merging**